test/e2e/pd.go

```diff
@@ -420,6 +421,81 @@ var _ = framework.KubeDescribe("Pod Disks", func() {
 		waitForPDDetach(disk1Name, host0Name)
 		waitForPDDetach(disk2Name, host0Name)
 	})
+
+	It("should be able to detach from a node which was deleted [Slow] [Disruptive]", func() {
+		framework.SkipUnlessProviderIs("gce")
+
+		initialGroupSize, err := GroupSize(framework.TestContext.CloudConfig.NodeInstanceGroup)
+		framework.ExpectNoError(err, "Error getting group size")
+
+		By("Creating a pd")
+		diskName, err := createPDWithRetry()
+		framework.ExpectNoError(err, "Error creating a pd")
+
+		host0Pod := testPDPod([]string{diskName}, host0Name, false, 1)
+		host1Pod := testPDPod([]string{diskName}, host1Name, false, 1)
+
+		containerName := "mycontainer"
+
+		defer func() {
+			By("Cleaning up PD-RW test env")
+			podClient.Delete(host0Pod.Name, api.NewDeleteOptions(0))
+			podClient.Delete(host1Pod.Name, api.NewDeleteOptions(0))
+			detachAndDeletePDs(diskName, []types.NodeName{host1Name})
+		}()
+
+		By("submitting host0Pod to kubernetes")
+		_, err = podClient.Create(host0Pod)
+		framework.ExpectNoError(err, fmt.Sprintf("Failed to create host0pod: %v", err))
+
+		framework.ExpectNoError(f.WaitForPodRunningSlow(host0Pod.Name))
+
+		testFile := "/testpd1/tracker"
+		testFileContents := fmt.Sprintf("%v", mathrand.Int())
+
+		framework.ExpectNoError(f.WriteFileViaContainer(host0Pod.Name, containerName, testFile, testFileContents))
+		framework.Logf("Wrote value: %v", testFileContents)
+
+		// Verify that disk shows up in node 0's volumeInUse list
+		framework.ExpectNoError(waitForPDInVolumesInUse(nodeClient, diskName, host0Name, nodeStatusTimeout, true /* should exist*/))
+
+		output, err := exec.Command("gcloud", "compute", "instances", "list").CombinedOutput()
+		Expect(true, strings.Contains(string(output), string(host0Name)))
+
+		By("deleting host0")
+
+		output, err = exec.Command("gcloud", "compute", "instances", "delete", string(host0Name), "--project="+framework.TestContext.CloudConfig.ProjectID, "--zone="+framework.TestContext.CloudConfig.Zone).CombinedOutput()
```

Comment:
Instead of writing your own cluster resizing routine, reuse the existing code in `e2e/resize_nodes.go`:

```GO
By(fmt.Sprintf("decreasing cluster size to %d", initialGroupSize-1))
err = ResizeGroup(group, initialGroupSize-1)
Expect(err).NotTo(HaveOccurred())
err = WaitForGroupSize(group, initialGroupSize-1)
Expect(err).NotTo(HaveOccurred())
err = framework.WaitForClusterSize(c, int(initialGroupSize-1), 10*time.Minute)
Expect(err).NotTo(HaveOccurred())
```

Maybe stick it in a common method `ResizeCluster(newSize int)`.

Also immediately prior to resizing make sure to call `defer...` that reverts the cluster to the original size (even if there is a failure). Again you can probably reuse the code from `AfterEach` in `ResizeCluster(newSize int)`